src/sidebar/components/selection-tabs.js

```diff
@@ -1,49 +1,212 @@
-    this.TAB_NOTES = uiConstants.TAB_NOTES;
-    this.TAB_ORPHANS = uiConstants.TAB_ORPHANS;
-
-    this.isThemeClean = settings.theme === 'clean';
-
-    this.enableExperimentalNewNoteButton =
-      settings.enableExperimentalNewNoteButton;
-
-    this.selectTab = function(type) {
-      store.clearSelectedAnnotations();
-      store.selectTab(type);
-    };
-
-    this.showAnnotationsUnavailableMessage = function() {
-      return (
-        this.selectedTab === this.TAB_ANNOTATIONS &&
-        this.totalAnnotations === 0 &&
-        !this.isWaitingToAnchorAnnotations
-      );
-    };
-
-    this.showNotesUnavailableMessage = function() {
-      return this.selectedTab === this.TAB_NOTES && this.totalNotes === 0;
-    };
-
-    this.showSidebarTutorial = function() {
-      return sessionUtil.shouldShowSidebarTutorial(session.state);
-    };
-  },
-  bindings: {
-    isLoading: '<',
-    isWaitingToAnchorAnnotations: '<',
-    selectedTab: '<',
-    totalAnnotations: '<',
-    totalNotes: '<',
-    totalOrphans: '<',
-  },
-  template: require('../templates/selection-tabs.html'),
+/**
+ *  Display name of the tab and annotation count.
+ */
+function Tab({
+  children,
+  count,
+  isWaitingToAnchor,
+  onChangeTab,
+  selected,
+  type,
+}) {
+  return (
+    <a
+      className={classnames('selection-tabs__type', {
+        'is-selected': selected,
+      })}
+      onClick={onChangeTab.bind(this, type)}
+      onTouchStart={onChangeTab.bind(this, type)}
+    >
+      <span>
+        {children}
+        {count > 0 && !isWaitingToAnchor && (
+          <span className="selection-tabs__count">{count}</span>
+        )}
+      </span>
+    </a>
+  );
+}
+Tab.propTypes = {
+  /**
+   * Child components.
+   */
+  children: propTypes.node.isRequired,
+  /**
+   * The total annotations for this tab.
+   */
+  count: propTypes.number.isRequired,
+  /**
+   * Are there any annotations still waiting to anchor?
+   */
+  isWaitingToAnchor: propTypes.bool.isRequired,
+  /**
+   * Callback when this tab is active with type as a parameter.
+   */
+  onChangeTab: propTypes.func.isRequired,
+  /**
+   * Is this tab currently selected?
+   */
+  selected: propTypes.bool.isRequired,
+  /**
+   * The type value for this tab. One of
+   * 'annotation', 'note', or 'orphan'.
+   */
+  type: propTypes.string.isRequired,
 };
+
+/**
+ *  Tabbed display of annotations and notes.
+ */
+
+function SelectionTabs({
+  isWaitingToAnchorAnnotations,
+  isLoading,
+  selectedTab,
+  totalAnnotations,
+  totalNotes,
+  totalOrphans,
+  settings,
+  session,
+}) {
+  // actions
+  const store = useStore(store => ({
+    clearSelectedAnnotations: store.clearSelectedAnnotations,
+    selectTab: store.selectTab,
+  }));
+
+  const isThemeClean = settings.theme === 'clean';
+
+  const selectTab = function(type) {
+    store.clearSelectedAnnotations();
+    store.selectTab(type);
+  };
+
+  const showAnnotationsUnavailableMessage = (() => {
```

Comment:
`showAnnotationsUnavailableMessage`, `showNotesUnavailableMessage` and `showSidebarTutorial` can all be values rather than immediately executed functions. The one valid case for using an IIFE that I've seen @lyzadanger use sometimes is when the value is complex to calculate and it is useful to split up into several statements. All of these functions just consist of a single `return` statement though.